tensorflow/contrib/bayesflow/python/ops/hmc_impl.py

```diff
@@ -308,7 +308,7 @@ def log_prob_beta(x):
 
 
 def kernel(step_size, n_leapfrog_steps, x, target_log_prob_fn, event_dims=(),
```

Comment:
To make your test pass, you'll have to do a few things.  The end result will be simpler.

First, add the import and a separate alias
```import tensorflow.python.ops.distributions import util as distribution_util```

Second, to `kernel()` and every function called by `kernel()`, you'll have to add a kwarg `seed` (make it the last kwarg), with default value `None`.

Third, every time a random number is generated, you need to add a "salted" version of `seed`.  E.g.
```m = random_ops.random_normal(x_shape, dtype=x.dtype, seed=distribution_util.gen_new_seed(seed, salt="m")```
Make sure the salt is unique for every call.  Also make sure that if one function calls another, a salted version of the seed is passed on.

Finally, inside your test, you can use the same seed for both calls to `kernel()`.  This should make things work.

Have fun!

Reply:
I'm not sure that's my definition of fun :D